### PR TITLE
Next step in connecting libdispatch and foundation builds

### PR DESCRIPTION
**Pull request by dgrove-oss**
Three small changes for building libdispatch and foundation together.
  (1) Put libdispatch into PRODUCTS before foundation
  (2) Pass path to swift down into libdispatch build
  (3) Pass paths to libdispatch down into foundation build

### CODE REVIEW

File: utils/build-script-impl [hunk 1]
```diff
@@ -1828,7 +1835,7 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_TOOLS_DEPLOYMENT_TARGETS[@]}"
                     autoreconf -fvi
                     popd
                     pushd "${LIBDISPATCH_BUILD_DIR}"
-                    "${LIBDISPATCH_SOURCE_DIR}"/configure --prefix="${INSTALL_DESTDIR}"/"${INSTALL_PREFIX}"
+                    "${LIBDISPATCH_SOURCE_DIR}"/configure --prefix="${INSTALL_DESTDIR}"/"${INSTALL_PREFIX}" --with-swift-sdk="${SWIFT_BUILD_PATH}"
```
Comment (gribozavr) [inline utils/build-script-impl]: "SDK" seems to be the wrong way to call the build directory.  It is usually called the "toolchain".  The SDK is the directory with the headers and frameworks.

**Comment by gribozavr** [comment]
LGTM modulo the SDK comment.

File: utils/build-script-impl (hunk 1 above)
Comment (dgrove-oss) [inline utils/build-script-impl]: ok.  thanks.  I didn't know that.  I'll make the change here and in the corresponding pull request to libdispatch.

**Comment by dgrove-oss** [comment]
changed to --with-swift-toolchain & squashed

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
Awesome work, @dgrove-oss. Very excited to see this happen.

On a related note, I'm expecting to make a similar change soon that establishes a dependency between XCTest and Foundation. The pull request that will introduce the dependency is https://github.com/apple/swift-corelibs-xctest/pull/43, the corresponding change to the build script is https://github.com/modocache/swift/commit/a17c05b630b472a410a2b3a525c9878998c84ce1. Just a heads up!

**Comment by drewcrawford** [comment]
LGTM.  I'm currently cherry-picking this commit into my binaries.

This is amazing, SO excited to have dispatch on Linux.

**Comment by dgrove-oss** [comment]
@gribozavr.  Hi, just pinging to see if this could be merged.  Looks like CI tests passed. Thanks!

**Comment by gribozavr** [comment]
@dgrove-oss Sorry for the delay!

**Pull request merged**